go-controller/pkg/config/config.go

```diff
@@ -631,6 +632,35 @@ func InitConfigWithPath(ctx *cli.Context, exec kexec.Interface, configFile strin
 	overrideFields(&CNI, &cfg.CNI)
 	overrideFields(&CNI, &cliConfig.CNI)
 
+	// values for token, cacert, kubeconfig, api-server may be found in several places.
+	// Take the first found when looking in this order: command line options, environment variables, files
+
+	if cliConfig.Kubernetes.Token == "" {
+		cliConfig.Kubernetes.Token = os.Getenv("K8S_TOKEN")
+	}
+	if cliConfig.Kubernetes.Token == "" {
+		if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token"); err == nil {
+			cliConfig.Kubernetes.Token = string(data)
+		}
+	}
+
+	if cliConfig.Kubernetes.CACert == "" {
+		cliConfig.Kubernetes.CACert = os.Getenv("K8S_CACERT")
+	}
+	if cliConfig.Kubernetes.CACert == "" {
+		if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"); err == nil {
+			cliConfig.Kubernetes.CACert = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+		}
+	}
+
+	if cliConfig.Kubernetes.Kubeconfig == "" {
+		cliConfig.Kubernetes.Kubeconfig = os.Getenv("K8S_KUBECONFIG")
```

Comment:
The convention is to just use `KUBECONFIG` as the environment variable here.

Reply:
OK